src/common/lib/hid_interface.c

```
@@ -35,122 +35,124 @@
-    const uint8_t* p = (const uint8_t*)report;
-    for (uint16_t i = 0; i < len && offset < sizeof(buffer) - 4; i++) {
-      offset += (size_t)snprintf(buffer + offset, sizeof(buffer) - offset, "%02X ", *p++);
-    }
-    
-    // Log at appropriate level based on result
-    if (result) {
-      LOG_DEBUG("%s\n", buffer);
-    } else {
-      LOG_ERROR("HID Report Send Failed (instance=%u, report_id=0x%02X, len=%u)\n", instance, report_id, len);
-      LOG_ERROR("%s\n", buffer);
+static inline bool hid_send_report(uint8_t instance, uint8_t report_id, void const* report,
```

Clang (14.0.6)

[warning] function 'hid_send_report' has cognitive complexity of 28 (threshold 25)

(readability-function-cognitive-complexity)